### PR TITLE
Reorganize theme resources

### PR DESCRIPTION
**Pull request by seadowg**
This blocks #3207 and #3225.

While focusing on our [typography rework](https://forum.opendatakit.org/t/reworking-collects-typography/20634) I've found that it feels like there is a lot of noise `theme.xml` and `color.xml` files as they both contain sets of colors and styles used for the app's Dark or Light theme. As we're going to be moving more attributes into these themes (typography attributes etc) it felt like the right time to break everything up a bit. I've tried to explain the larger changes in commit messages.

#### What has been done to verify that this works as intended?

Ran tests locally and eyeballed app to make sure it didn't look like anything had changed.

#### Why is this the best possible solution? Were any other approaches considered?

This is a potentially very subjective approach as I could see people feeling that having more centralization of theme, style and color values would have been a better way to go. Personally though, I found it easier to understand everything once I broke the different light and dark themes into mostly separate files and then also split any component specific themes into their own files. Definitely interested to hear thoughts. 

I think a development on this approach would be to de duplicate color definitions that are being using in theme attributes using defined material colors (like `@color/material_grey_500` etc). I wanted to get feedback on this before going down that route though - feels like it might be a monotonous task...

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

There shouldn't be any visible effect. Would be good to double check that dark/light theme switches still seem all good to double check no colors have switched sides on accident.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/res/values/permission_dialog_theme.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="PermissionAlertDialogTheme" parent="Theme.AppCompat.Light.Dialog.Alert">
```
Comment (lognaturel) [inline collect_app/src/main/res/values/permission_dialog_theme.xml]: `PermissionAlertDialogTheme` should probably be renamed as other theme names have been.

**Comment by lognaturel** [comment]
I do think the split makes things easier to understand. I think it would be helpful to have a comment at the top of the base light and dark themes that quickly says that activity/dialog themes should go in their own files.

Another thing this PR does is change/establish naming conventions. @seadowg does the Android project document a naming convention? I think it would be good to follow some existing convention and document it in the contribution guide.

**Comment by seadowg** [comment]
@lognaturel makes sense. I will:

- [x]  look at adding more comments 
- [x] have a dig into if there are actually any "official" naming conventions for these things and add to our documentation

File: collect_app/src/main/res/values/permission_dialog_theme.xml (hunk 1 above)
Comment (seadowg) [reply collect_app/src/main/res/values/permission_dialog_theme.xml]: Done!

**Comment by shobhitagarwal1612** [comment]
The changes look really neat and well thought! Really liking the new code structure ✨

**Comment by seadowg** [comment]
@lognaturel sorry I left the `needs-discussion` label on by mistake after this was discussed. This looking good to go?

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
> I think a development on this approach would be to de duplicate color definitions that are being using in theme attributes using defined material colors (like @color/material_grey_500 etc)

Sorry, I'm not really following. You're talking about giving names to colors that Collect defines? Using colors that are already defined somewhere? Something else...?

**Comment by seadowg** [comment]
@lognaturel wow reading that back was hard. Not my best writing. I meant instead of having this in our `light_theme.xml`:

```xml
<item name="primaryTextColor">#de000000</item>
```

We'd have:

```xml
<item name="primaryTextColor">@color/material_black_200</item>
```

And having the different colors we use (but named after the color they are, not what they are being used for when we define the color) in `colors.xml`. This avoids us having a color we add for say "errors" being added as `@color/error` but then being used all over the app for different things.

**Comment by lognaturel** [comment]
I see, that makes a lot of sense. Seems worth doing as you're touching those colors anyway. I don't personally mind a mix of colors that are just named according to their usage semantics and some that have both levels of naming -- usage semantics and color description.

**Comment by seadowg** [comment]
@lognaturel we'll have to relook at colors when we shift to a [Material Components Theme](https://material.io/develop/android/theming/color/) so I'd prefer to do that then I think. Does that make sense?

**Comment by bognasoldev89** [comment]
Hi, @seadowg is it possible to update this branch as close to master as possible?
I am testing this pr right now and it would make my job much easier - for example, maps settings are out of date and it makes it harder to compare views.

**Comment by seadowg** [comment]
@bognasoldev89 done!

**Comment by seadowg** [comment]
Unit test fail will be fixed by #3302

**Comment by bognasoldev89** [comment]
*Tested with success*

Verified on Android: 4.2, 4.4, 5.1, 6.0, 7.0, 8.1

Verified cases:
- Reorganize theme resources
- Changing themes
- Rotating phone
- Blocking phone
- Minimizing the app
- Going through different elements of the app

As was said, I was looking for visual inconsistencies before and after reorganizing and I cannot say I found any. ODK Collect did not crash during testing this PR.

Some examples before and after:

| | |
| --- | --- |
| image | image (1) |
| Screenshot_20190813-152305 | Screenshot_20190813-155455 |
| Screenshot_20190813-152226 | Screenshot_20190813-152733 |


@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Pull request merged**